### PR TITLE
Set association

### PR DESCRIPTION
**Pull request by gids-dadi**
In the branch of the this PR, I created
- has-many/belongs-to association between Classroom and Student
- many-to-many Person and Book

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Rental
+  def initialize(date)
+    @date = date
+  end
+end
```
Comment (Deepakdanger) [inline rental.rb]: - [ ] In rental, we have initialize, book & person, Kindly include that also.
```
class Rental
  attr_accessor :date, :book, :person
  .
  
  def initialize(date, book, person)
    .
    .

    @book = book
    # *The rental is adding itself to the @book achieving the 'belongs-to' association
    book.rentals << self

    @person = person
    # *The rental is adding itself to the @person the 'belongs-to' association
    person.rentals << self
  end
end
```

File: person_class.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
-class Person
+require_relative './nameable'
+require_relative 'rental'
+
+class Person < Nameable
```
Comment (Deepakdanger) [inline person_class.rb:4]: - [ ] Here we have to include this method also:-
```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @gids-dadi, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by medaminedev66 — APPROVED** [review]
Hi @gids-dadi 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your code is clean

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**